modules/registrars/openprovider/src/DomainSyncAutoRenew.php

```diff
@@ -0,0 +1,525 @@
+
+            Activity::log('update_domain_empty_next_due_date', $activity_data);
+        }
+
+        $this->printDebug('DONE PROCESSING EMPTY NEXT DUE DATES');
+    }
+
+    /**
+     * Process the Domain status setting
+     *
+     * @param  string $status Set the domain status
+     * @return void
+     **/
+    private function process_domain_status($status = null)
+    {
+        if($status == 'Cancelled' || $status == 'Expired' || $status == 'Transferred Away')
+        {
+            // Nothing to do.
+            if($this->objectDomain->status == $status)
+                return;
+
+            $this->update_domain_data['status'] = $status;
+
+            /**
+             * Setup an hook to log the domain status
+             */
+            $activity_data = [
+                'old_status'      => $this->objectDomain->status,
+                'new_status'      => $status
+            ];
+
+            $this->update_executed_logs[] = ['activity' => 'update_domain_status', 'data' => $activity_data];
+
+            return;
+        }
+
+        $op_domain_status = api_domain::convertOpStatusToWhmcs($this->op_domain['status']);
+
+        if($op_domain_status != false)
+        {
+            // Check if the status matches
+            if($this->objectDomain->status != $op_domain_status)
+            {
+                if($this->objectDomain->status == 'Pending Transfer' && $op_domain_status == 'Active')
+                {
+                    if($this->objectDomain->check_renew_domain_setting_upon_completed_transfer() == true)
+                    {
+                        $this->apiHelper->renewDomain($this->op_domain['id'], $this->objectDomain->registrationperiod);
+
+                        // Fetch updated information
+                        $this->op_domain             =    $this->apiHelper->getDomain($this->op_domain_obj);
+                    }
+
+                    // Since this is a transfer, we will always update the expiration date. We ignore the configuration setting
+                    // since this is an initial transfer.
+                    $this->process_expiry_date();
+                }
+
+                // It does not, let's update the data.
+                $this->update_domain_data['status'] = $op_domain_status;
+
+                /**
+                 * Setup an hook to log the domain status
+                 */
+                $activity_data = [
+                    'old_status'      => $this->objectDomain->status,
+                    'new_status'      => $op_domain_status
+                ];
+
+                $this->printDebug('PROCESSING DOMAIN STATUS CHANGE ' . $this->objectDomain->domain .' ( ' . $this->objectDomain->status . ' => ' . $op_domain_status . ')');
+
+                $this->update_executed_logs[] = ['activity' => 'update_domain_status', 'data' => $activity_data];
+
+            }
+
+        }
+        else
```

Comment:
I can see most of the places, you added curly braces in new lines, for consitions better to use same like you did try catch